gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/BatchImpl.java

```diff
@@ -16,65 +16,53 @@
 
 package com.google.gcloud.datastore;
 
-import com.google.api.services.datastore.DatastoreV1;
-import com.google.common.base.Function;
-import com.google.common.collect.Lists;
-import com.google.gcloud.datastore.BatchOption.ForceWrites;
-
+import java.util.Iterator;
+import java.util.LinkedList;
 import java.util.List;
-import java.util.Map;
 
 
 class BatchImpl extends BaseDatastoreBatchWriter implements Batch {
 
   private final DatastoreImpl datastore;
-  private final boolean force;
 
   static class ResponseImpl implements Batch.Response {
 
-    private final DatastoreV1.CommitResponse response;
+    private final com.google.datastore.v1beta3.CommitResponse response;
+    private final int numAutoAllocatedIds;
 
-    ResponseImpl(DatastoreV1.CommitResponse response) {
+    ResponseImpl(com.google.datastore.v1beta3.CommitResponse response, int numAutoAllocatedIds) {
       this.response = response;
+      this.numAutoAllocatedIds = numAutoAllocatedIds;
     }
 
     @Override
     public List<Key> generatedKeys() {
-      return Lists.transform(response.getMutationResult().getInsertAutoIdKeyList(),
-          new Function<DatastoreV1.Key, Key>() {
-            @Override public Key apply(DatastoreV1.Key keyPb) {
-              // TODO(ajaykannan): fix me!
-              //return Key.fromPb(keyPb);
-              return Key.builder(null).build(); // TODO(ajaykannan): fix me!
-            }
-          });
+      Iterator<com.google.datastore.v1beta3.MutationResult> results = 
+          response.getMutationResultsList().iterator();
+      List<Key> generated = new LinkedList<Key>();
+      for (int i = 0; i < numAutoAllocatedIds; i++) {
+        generated.add(Key.fromPb(results.next().getKey()));
+      }
+      return generated;
     }
   }
 
-  BatchImpl(DatastoreImpl datastore, BatchOption... options) {
+  BatchImpl(DatastoreImpl datastore) {
     super("batch");
     this.datastore = datastore;
-    Map<Class<? extends BatchOption>, BatchOption> optionsMap = BatchOption.asImmutableMap(options);
-    if (optionsMap.containsKey(ForceWrites.class)) {
-      force = ((ForceWrites) optionsMap.get(ForceWrites.class)).force();
-    } else {
-      force = datastore.options().force();
-    }
   }
 
   @Override
   public Batch.Response submit() {
     validateActive();
-    DatastoreV1.Mutation.Builder mutationPb = toMutationPb();
-    if (force) {
-      mutationPb.setForce(force);
```

Comment:
We are removing the ability to force a write while in read-only using the API. @Alfus any extra thoughts?